coriolis/tests/minion_manager/rpc/data/get_minion_pool_refresh_flow.yaml

```diff
@@ -0,0 +1,159 @@
+# perform health checks for powered on machines
+# skip health checks for non-powered on machines
+- config:
+    minion_pool:
+      id: minion_pool1
+      minimum_minions: 4
+      minion_retention_strategy: delete
+      minion_machines:
+        - id: minion1
+          allocation_status: AVAILABLE
+          power_status: POWERED_ON
+        - id: minion2
+          allocation_status: ERROR
+          power_status: POWERED_ERROR
+        - id: minion3
+          allocation_status: AVAILABLE
+          power_status: POWERED_ON    
+        - id: minion4
+          allocation_status: ERROR_DEPLOYING
+          power_status: POWERED_ON    
+  expect:
+    result:
+      flow_tasks: 
+        - pool-minion_pool1-machine-minion1-healthcheck
+        - pool-minion_pool1-machine-minion3-healthcheck
+        - pool-minion_pool1-machine-minion4-healthcheck
+      # "include" are the db calls that must be present
+      # "exclude" are the db calls that must not be present
+      db_calls:
+        include:
+          - set_minion_machine_allocation_status: 
+              id: minion1
+              allocation_status: HEALTHCHECKING
+          - set_minion_machine_allocation_status:
+              id: minion3
+              allocation_status: HEALTHCHECKING
+          - set_minion_machine_allocation_status:
+              id: minion4
+              allocation_status: HEALTHCHECKING
+        exclude:
+          - set_minion_machine_allocation_status:
+              id: minion2
+              allocation_status: HEALTHCHECKING
+
+# power off machines that are not needed
+- config:
+    minion_pool:
+      id: minion_pool1
+      # minimum_minions 1 means we have too many minions
+      minimum_minions: 1
+      minion_max_idle_time: 0
```

Comment:
You set this, but never set `last_used_at` property on the machines, thus assuming all machines are expired and need poweroff/deallocation. Please also test that unexpired machines are not touched. 
I recommend adding a `last_used_at` attribute, which is a list that you can pass to `datetime.datetime`, therefore in your test code you will have:
```python
minion_pool = testutils.DictToObject(config.get("minion_pool"), {})
if config.get('minion_pool', {}).get('last_used_at'):
    minion_pool.last_used_at = datetime.datetime(*minion_pool.last_used_at)
```